From: Eric Sandeen <sandeen@redhat.com>
Date: Fri, 17 Apr 2009 13:33:31 -0500
Subject: [fs] export set_task_ioprio
Message-id: 49E8CB7B.1000007@redhat.com
O-Subject: [RHEL5.4 PATCH 1/6] export set_task_ioprio
Bugzilla: 485315
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Josef Bacik <josef@redhat.com>

This is for:
Bug 485315 -  ext4 kernelspace rebase for RHEL5.4

Part of the following upstream commit:

From: Theodore Ts'o <tytso@mit.edu>
Date: Tue, 6 Jan 2009 03:46:26 +0000 (-0500)
Subject: ext4: Add mount option to set kjournald's I/O priority
X-Git-Tag: v2.6.29-rc1~40^2~5
X-Git-Url: http://git.engineering.redhat.com/?p=linux-2.6.git;a=commitdiff_plain;h=b3881f74b31b7d47d0f1c4d89ac3e7f0b9c05e3e

ext4: Add mount option to set kjournald's I/O priority

Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
Cc: Jens Axboe <jens.axboe@oracle.com>

diff --git a/fs/ioprio.c b/fs/ioprio.c
index 78b1dea..c98b4b7 100644
--- a/fs/ioprio.c
+++ b/fs/ioprio.c
@@ -26,7 +26,7 @@
 #include <linux/syscalls.h>
 #include <linux/security.h>
 
-static int set_task_ioprio(struct task_struct *task, int ioprio)
+int set_task_ioprio(struct task_struct *task, int ioprio)
 {
 	int err;
 	struct io_context *ioc;
@@ -53,6 +53,7 @@ static int set_task_ioprio(struct task_struct *task, int ioprio)
 	task_unlock(task);
 	return 0;
 }
+EXPORT_SYMBOL_GPL(set_task_ioprio);
 
 asmlinkage long sys_ioprio_set(int which, int who, int ioprio)
 {
diff --git a/include/linux/ioprio.h b/include/linux/ioprio.h
index 8e2042b..63420f2 100644
--- a/include/linux/ioprio.h
+++ b/include/linux/ioprio.h
@@ -61,4 +61,6 @@ static inline int task_nice_ioprio(struct task_struct *task)
  */
 extern int ioprio_best(unsigned short aprio, unsigned short bprio);
 
+extern int set_task_ioprio(struct task_struct *task, int ioprio);
+
 #endif
